src/Model/Resolver/Menu.php

```diff
@@ -84,43 +83,48 @@ public function resolve(
         ResolveInfo $info,
         array $value = null,
         array $args = null
-    ): Value {
-        $result = function () {
-            return null;
-        };
-
-        if (isset($args['identifier'])) {
-            $menu = $this->menuCollectionFactory->create();
-
-            /** Updated with identifier filtering */
-            $menu->addFieldToFilter('identifier', $args['identifier'])->load();
+    )
+    {
+        $identifier = $args['identifier'];
 
-            $menuData = $menu->getFirstItem()->getData();
+        $menu = $this->menuFactory->create();
+        $this->menuResourceModel->load($menu, $identifier);
```

Comment:
We could pass $args['identifier'] directly.

Reply:
We use `$identifier` variable later on, when throwing exception.